### PR TITLE
Allow usage of jemalloc in ddprof

### PR DESCRIPTION
**Pull request by r1viollet**
# What does this PR do?

- Install jemalloc on alpine
- Remove prints of jemalloc stats
- Define jemalloc as the default and use a function to retrieve this from downstream build pipeline

# Motivation

Solve performance issue of musl allocator

### CODE REVIEW

**Comment by r1viollet** [comment]
image

File: tools/style-check.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 cd "$SCRIPT_DIR/.."
 
 # Find most recent clang-format, defaulting to an unqualified default
-CLANG_FORMAT=$(command -v clang-format{-13,-12,-11,-10,-9,} | head -n1)
+CLANG_FORMAT=$(command -v clang-format{-15,-13,-12,-11,-10,-9,} | head -n1)
```
Comment (sanchda) [inline tools/style-check.sh:13]: Good call, 15 hit earlier this year.

**Review by sanchda — APPROVED** [review]
LGTM.  I can't predict any difficulties from using jemalloc and we know from previous experience that it actually has a tremendous impact on RSS increases from libdwfl.  Good idea, thank you!

**Comment by nsavoire** [comment]
When a library is available both as a shared and static library, gcc will prefer linking with the shared one except if `-static` option is used. As a result, `jemalloc` will be linked dynamically on ubuntu, which is not the desired behaviour.

**Comment by sanchda** [comment]
> `jemalloc` will be linked dynamically on ubuntu which might not be the desired behaviour.

Good call, @nsavoire, since we can't assume the user will have jemalloc installed.

File: tools/style-check.sh (hunk 1 above)
Comment (nsavoire) [inline tools/style-check.sh:13]: Is `clang-format-14`  ostracised :wink: ?

**Pull request merged**